prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -87,6 +95,40 @@ object FhirPathUtils : Logging {
         bundle: Bundle,
         expressionNode: ExpressionNode
     ): String {
-        return defaultPathEngine.evaluateToString(appContext, focusResource, bundle, bundle, expressionNode)
+        val evaluated = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, expressionNode)
+        if (evaluated.size == 0) return ""
+        if (evaluated.size > 1 || !evaluated[0].isPrimitive) {
+            // Should only be evaluating primitives here
+            throw IllegalStateException()
+        }
+        return when (val primitive = evaluated[0]) {
+            is InstantType, is DateTimeType -> convertDateTimeToHL7(primitive)
+            is DateType -> convertDateToHL7(primitive)
+            is TimeType -> convertTimeToHL7(primitive)
+            else -> defaultPathEngine.convertToString(primitive)
+        }
+    }
+    fun convertDateTimeToHL7(dateTime: Base): String {
+        val pattern = "YYYYMMddHHmmss.SSSSxxxx"
+        val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern(pattern)
+        val zonedDateTime: ZonedDateTime = ZonedDateTime.parse(dateTime.dateTimeValue().valueAsString)
+
+        return zonedDateTime.format(formatter)
+    }
+
+    fun convertTimeToHL7(timeType: TimeType): String {
+        val pattern = "HHmmss.SSSS"
+        val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern(pattern)
+        val localTime: LocalTime = LocalTime.parse(timeType.valueAsString)
```

Comment:
Can you use the time itself, instead of converting it first to a string?  E.g., date times can generate a Date object, which then you can use to generated the formatted string.

Reply:
From what I can tell, the only way to go from a FHIR DateType, TimeType, or DateTimeType object to its equivalent kotlin object is to get the string value first. I haven't found a way to directly format the FHIR objects